orbit/pkg/installer/installer.go

```diff
@@ -70,20 +71,30 @@ type Runner struct {
 	scriptsEnabled func() bool
 
 	osqueryConnectionMutex sync.Mutex
+
+	rootDirPath string
 }
 
-func NewRunner(client Client, socketPath string, scriptsEnabled func() bool) *Runner {
+func NewRunner(client Client, socketPath string, scriptsEnabled func() bool, rootDirPath string) *Runner {
 	r := &Runner{
 		OrbitClient:               client,
 		osquerySocketPath:         socketPath,
 		scriptsEnabled:            scriptsEnabled,
 		installerExecutionTimeout: pkgscripts.MaxHostSoftwareInstallExecutionTime,
+		rootDirPath:               rootDirPath,
 	}
 
 	return r
 }
 
 func (r *Runner) Run(config *fleet.OrbitConfig) error {
+	if runtime.GOOS == "darwin" {
+		if config.Notifications.RunSetupExperience && update.CanRun(r.rootDirPath, "swiftDialog", update.SwiftDialogMacOSTarget) {
+			log.Debug().Msg("exiting software installer config runner early during setup experience: swiftDialog is not installed")
+			return nil
```

Comment:
Do I understand correctly that this will be re-run later, at regular intervals, so that eventually swift dialog will be installed and this will proceed?